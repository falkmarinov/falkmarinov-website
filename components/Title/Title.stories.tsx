import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';

import { Title } from './Title';

export default {
  title: 'Typography/Title',
  component: Title,
  args: {
    children: 'Lorem Ipsum',
  },
} as ComponentMeta<typeof Title>;

const Template: ComponentStory<typeof Title> = (args) => <Title {...args} />;

export const Default = Template.bind({});
Default.args = {};
