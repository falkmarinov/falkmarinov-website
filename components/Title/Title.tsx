import React from 'react';

import cn from 'classnames';

interface TitleProps {
  /**
   * Title text
   */
  children: string;
}

/**
 * Title typography
 */
export const Title = ({ children }: TitleProps) => {
  const classNames = cn(
    'text-3xl sm:text-4xl lg:text-5xl tracking-[0.16em] font-bold'
  );

  return <h1 className={classNames}>{children}</h1>;
};
