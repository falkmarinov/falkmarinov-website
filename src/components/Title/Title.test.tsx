import { render } from '@testing-library/react';

import { Title } from './Title';

describe('<Title />', () => {
  describe.each([['First Title'], ['Second Title']])(
    'should render different titles',
    (text) => {
      it(text, () => {
        const { container } = render(<Title>{text}</Title>);
        expect(container).toMatchSnapshot();
      });
    }
  );
});
