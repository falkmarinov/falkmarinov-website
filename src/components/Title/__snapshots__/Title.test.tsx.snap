// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render different titles First Title 1`] = `
<div>
  <h1
    class="text-3xl sm:text-4xl lg:text-5xl tracking-[0.16em] font-bold"
  >
    First Title
  </h1>
</div>
`;

exports[`<Title /> should render different titles Second Title 1`] = `
<div>
  <h1
    class="text-3xl sm:text-4xl lg:text-5xl tracking-[0.16em] font-bold"
  >
    Second Title
  </h1>
</div>
`;
